File: src/main/java/com/sfh/pokeRogueBot/phase/PhaseProvider.java
package com.sfh.pokeRogueBot.phase;

import lombok.Getter;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@Getter
public class PhaseProvider {

    private final List<Phase> phases;

    public PhaseProvider(List<Phase> phases) {
        this.phases = phases;
    }

    public Phase fromString(String phaseAsString) {
        return phases.stream()
                .filter(phase -> phase.getPhaseName().equals(phaseAsString))
                .findFirst()
                .orElse(null);
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/phase/PhaseProvider.kt
package com.sfh.pokeRogueBot.phase

import com.sfh.pokeRogueBot.model.exception.UnsupportedPhaseException
import org.springframework.stereotype.Component

@Component
class PhaseProvider(private val phases: Set<Phase>) {

    fun fromString(phaseAsString: String): Phase {
        val phase = phases.find { phase -> phase.phaseName == phaseAsString }
        if(null != phase) {
            return phase
        }
        throw UnsupportedPhaseException(phaseAsString)
    }
}